### DISCUSSION TITLE
How best to customize mail from header "From: Default Org Team <hi@defaultorg.com>"

### DISCUSSION THREAD

**Discussion by BenSturmfels** [houdiniproject/houdini#1116] [Q&A]
Hi Folks,
I see that when I click "resend a payment receipt to yourself", the email generated has the header "From: Default Org Team <hi@defaultorg.com>". I see that this is configured in `config/application.rb`, so I can obviously customize it there, but is that the recommended approach - maintaining untracked changes to to `config/application.rb`?

Would it perhaps make sense for the outgoing mail to come from the "org email" that's set up under Settings?

**Comment by wwahammy** [comment]
Hi @BenSturmfels:

Generally, to date, receipts come from a single email address for an Houdini instance and it would be set in the `config/houdini_config.rb` at `config.houdini.hoster.support_email`. This value defaults to whatever is in `config/application.rb`. At runtime, you can get this value by calling  `Houdini.hoster.support_email`.

The reasoning for having a single receipt email is mostly historical: if someone runs an Houdini instance and allows external nonprofits to join, it's difficult to send using the nonprofit's own domain as you won't have the proper SPF and DKIM records set up.

If one didn't worry about that OR an Houdini instance owner set up the appropriate DNS records for each nonprofit that has a receipt, then one could use the org email. That said, the org email is an optional property under settings and it's really mostly there for displaying on the Nonprofit's main donation page on the instance.

I'm open to the idea of having nonprofits be able to set their own receipt email addresses. We'd have to have a way to configure whether to allow that and we might not want to have the "org email" value do double duty but it's something we can think about.

**Reply by BenSturmfels** [reply]
Thanks Eric, that makes sense. When overriding the default `config.houdini.hoster.support_email`, is `config/houdini_config.rb` literally the right place to put this change on a production system? (not committed to version control)

**Comment by RamtinKosari** [comment]
When it comes to customizing email headers and configurations, it's generally not recommended to directly modify the config/application.rb file or any other core application files. Making untracked changes to core files can make it challenging to maintain and update your application in the future, as well as potentially introduce conflicts when merging updates from the upstream source.

Instead, a more recommended approach is to use configuration options or settings provided by the application or framework itself. In this case, if you want the outgoing email to come from the "org email" set up under Settings, you should look for a specific configuration option or setting related to email headers or sender information.

Here are a few steps you can follow to achieve this:

Check the application's documentation: Review the documentation or guides provided by the application or framework you're using. Look for information on how to customize email settings, including the sender address or header information.

Look for configuration files: Check if there are dedicated configuration files related to email settings. These files might provide options to specify the sender address or other email-related configurations. It's possible that the application has a separate configuration file specifically for email settings, rather than modifying the config/application.rb file.

Explore application-specific settings: Check the application's settings or configuration pages. Look for options related to email settings, such as the sender address or organization email. If there's a setting available for the "org email," you can update that setting to reflect the desired sender address for outgoing emails.

Use environment variables or configuration files: If there's no direct setting available, consider using environment variables or external configuration files. These can provide a flexible way to override default settings without modifying core files. You can set up an environment variable or configuration file specifically for the sender address and update the email configuration code to read from that variable or file.

By following these steps, you can customize the sender address or header information for outgoing emails without modifying core application files. This approach allows you to maintain your customizations separately and makes it easier to update the application in the future.

**Reply by BenSturmfels** [reply]
Heh, thanks chatbot. ;)
